common4j/src/main/com/microsoft/identity/common/java/interfaces/ICommonComponents.java

```diff
@@ -41,4 +42,44 @@
      * Get an encryption manager for storage layer.
      */
     IKeyAccessor getStorageEncryptionManager(@Nullable final ITelemetryCallback telemetryCallback);
+
+    /**
+     * Retrieve a name-value store with a given identifier.
+     *
+     * @param storeName The name of a new KeyValue store.
+     * @param clazz     The class of values in the name value store.
+     * @return a INameValueStorage instance based around data stored with the same storeName.
+     */
+    <T> INameValueStorage<T> getNameValueStore(String storeName, Class<T> clazz);
+
+    /**
+     * Retrieve a name-value store with a given identifier.
+     *
+     * @param storeName The name of a new KeyValue store. May not be null.
+     * @param helper    The key manager for the encryption.  May be null.
+     * @param clazz     The class of values in the name value store. May not be null.
+     * @return a INameValueStorage instance based around data stored with the same storeName.
+     */
+    <T> INameValueStorage<T> getEncryptedNameValueStore(String storeName, IKeyAccessor helper, Class<T> clazz);
+
+    /**
+     * Get a generic encrypted ISharedPreferencesFileManager with a given identifier.
+     *
+     * @param storeName The name of a new KeyValue store. May not be null.
+     * @param helper    The key manager for the encryption.  May not be null.
+     */
+    ISharedPreferencesFileManager getEncryptedFileStore(String storeName, IKeyAccessor helper);
+
+    /**
+     * Get a generic encrypted ISharedPreferencesFileManager with a given identifier.
+     *
+     * @param storeName The name of a new KeyValue store. May not be null.
+     */
+    ISharedPreferencesFileManager getFileStore(String storeName);
+
+    /**
+     * Unwrap the platform context being shadowed by this interface.
+     * @return the underlying implementation of the platform context that this is hiding.
```

Comment:
I think this should never exist.  IMO this is a code smell (or a result of one).

as in, we generally use Context to acquire `something else` (and this varies from platform to platform).

It's better to expose those `things` instead.

Reply:
Sure, but this is a way to do it without needing to do everything at once.

Reply:
I've dropped this.

Reply:
Is this still WIP? (You mentioned you dropped it but it's still here.

Reply:
I am very confused.  I definitely removed every usage, but somehow this got left. Removing it.